cmd/clusterctl/client/cluster/upgrader_test.go

```diff
@@ -182,7 +227,67 @@ func Test_providerUpgrader_Plan(t *testing.T) {
 					},
 				},
 				{ // one upgrade plan with the latest releases in the current contract
-					Contract: test.CurrentCAPIContract,
+					Contract: currentContractVersion,
+					Providers: []UpgradeItem{
+						{
+							Provider:    fakeProvider("cluster-api", clusterctlv1.CoreProviderType, "v1.0.0", "cluster-api-system"),
+							NextVersion: "v2.0.0",
+						},
+						{
+							Provider:    fakeProvider("infra", clusterctlv1.InfrastructureProviderType, "v2.0.0", "infra-system"),
+							NextVersion: "v3.0.0",
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Upgrade for compatible contract, current contract", // upgrade plan should report unsupported options
```

Comment:
Is this "upgrade plan should report unsupported options" comment outdated? (and a similar one below in this test case)

Reply:
The comment is still valid
Upgrade plan should give you all options. Then might be that you have to use a newer or older release of clusterctl if you want to pick an option about a contract version different than the one supported by this release

Reply:
Yup. But is this actually unsupported if it's about a compatible verison?

Reply:
It is unsupported (you cannot use this version of clusterctl for it).
Compatible version are only about providers, not about the management cluster or the core provider

Reply:
Okay got it. thx

Reply:
I thought unsupported was referring to the provider versions in the plans, but it is referring to the contract version